cpp/src/io/utilities/file_io_utilities.hpp

```diff
@@ -270,6 +294,10 @@ class cufile_output_impl final : public cufile_output {
   {
     CUDF_FAIL("Only used to compile without cufile library, should not be called");
   }
+  std::future<void> write_async(void const* data, size_t offset, size_t size) override
```

Comment:
I expected clang-format to add a newline between these definitions. If not, this comment can be ignored.